Update config-settings.rst (#3670)
mattermost/docs: source/administration/config-settings.rst

| This feature's ``config.json`` setting is ``"QueryTimeout": 30`` with numerical input.                                  |
+-------------------------------------------------------------------------------------------------------------------------+

DisableDatabaseSearch
^^^^^^^^^^^^^^^^^

> 

Reviewer: ```suggestion
**True:** Disables the use of the database to perform searches. Should only be used when other `search engines  <https://mattermost.com/pl/default-search-engine>`_ are configured.

**False:** Database search is not disabled. 

```

+-------------------------------------------------------------------------------------------------------------------------+

> | This feature's ``config.json`` setting is ``"QueryTimeout": true`` with              .                                  |

Reviewer: ```suggestion
| This feature's ``config.json`` setting is ``"DisableDatabaseSearch  ": false`` with options ``true`` and ``false``.                               |
```

| This feature's ``config.json`` setting is ``"QueryTimeout": 30`` with numerical input.                                  |
+-------------------------------------------------------------------------------------------------------------------------+

DisableDatabaseSearch
^^^^^^^^^^^^^^^^^
**True:** Disables the use of the database to perform searches. Should only be used when other `search engines  <https://mattermost.com/pl/default-search-engine>`_ are configured.

**False:** Database search is not disabled. 

+-------------------------------------------------------------------------------------------------------------------------+

> | This feature's ``config.json`` setting is ``"DisableDatabaseSearch  ": false`` with options ``true`` and ``false``.     |

Reviewer 2: nit:

```suggestion
| This feature's ``config.json`` setting is ``"DisableDatabaseSearch": false`` with options ``true`` and ``false``.       |
```

| This feature's ``config.json`` setting is ``"EnableLatex": false`` with options ``true`` and ``false``.                                                              |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Enable Local Mode
^^^^^^^^^^^^^^^^^^^^^^^^^

**True**: Enables local mode for mmctl.

**False**: Prevents local mode for mmctl.

+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

> | This feature's ``config.json`` setting is ``"EnableLocalMode": true`` with options ``true`` and ``false``.                                                           |

Reviewer 2: `EnableLocalMode` is `false` by default

```suggestion
| This feature's ``config.json`` setting is ``"EnableLocalMode": false`` with options ``true`` and ``false``.                                                          |
```

+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Enable Local Mode Socket Location
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The path for the socket that the server will create for mmctl to connect and communicate through local mode.  When connecting with mmctl through local mode, you will be required to specify not only the ``--local flag``, but the ``--local-socket-path`` ``/PATH/TO/SOCKET`` as well.

Reviewer 2: ```suggestion
The path for the socket that the server will create for mmctl to connect and communicate through local mode. If the default value for this key is changed, you will need to point mmctl to the new socket path when in local mode, using the ``--local-socket-path /new/path/to/socket`` flag in addition to the ``--local`` flag.
```

| This feature's ``config.json`` setting is ``"Trace": ""`` with string input.                          |
+-------------------------------------------------------------------------------------------------------+

> Bleve Settings

Reviewer: ```suggestion
Bleve Settings (Experimental)
```

~~~~~~~~~~~~~~~

IndexDir
^^^^^^^^^^^^^^^^^^^^^^^

EnableIndexing
^^^^^^^^^^^^^^^^^^^^^^^

EnableSearching
^^^^^^^^^^^^^^^^^^^^^^^

EnableAutocomplete
^^^^^^^^^^^^^^^^^^^^^^^

BulkIndexingTimeWindowSeconds
^^^^^^^^^^^^^^^^^^^^^^^

> 

Reviewer: ```suggestion
Determines the maximum time window for a batch of posts being indexed by the Bulk Indexer. This setting serves as a performance optimization for installs with over ~10 million posts in the database. Approximate this value based on the average number of seconds for 2,000 posts to be added to the database on a typical day in production. Setting this value too low will cause Bulk Indexing jobs to run slowly.

+-----------------------------------------------------------------------------------------------------------+
| This feature's ``config.json`` setting is ``"BulkIndexingTimeWindowSeconds:3600  "`` with numerical input.                                                         |
+----------------------------------------------------------------------------------------------------------
```

| This feature's ``config.json`` setting is ``"Trace": ""`` with string input.                          |
+-------------------------------------------------------------------------------------------------------+

Bleve Settings (Experimental)
~~~~~~~~~~~~~~~

> IndexDir

Author: ```suggestion
Index Dir
```

^^^^^^^^^^^^^^^^^^^^^^^
Directory path to use for storing bleve indexes.

+-----------------------------------------------------------------------------------------------------------+
| This feature's ``config.json`` setting is ``"IndexDir "`` with string input.                              |
+-----------------------------------------------------------------------------------------------------------+

> EnableIndexing

Author: ```suggestion
Enable Indexing
```

^^^^^^^^^^^^^^^^^^^^^^^
**True:** The indexing of new posts occurs automatically. Search queries will use database search until "Enable Bleve for search queries" is enabled.

**False:** The indexing of new posts does not occur automatically. 

+------------------------------------------------------------------------------------------------------------+
| This feature's ``config.json`` setting is ``"EnableIndexing": false`` with options ``true`` and ``false``. |
+------------------------------------------------------------------------------------------------------------+

> EnableSearching

Author: ```suggestion
Enable Searching
```

^^^^^^^^^^^^^^^^^^^^^^^
**True:** Search queries will use bleve search.

**False:** Search queries will not use bleve search. 
+--------------------------------------------------------------------------------------------------------------+
| This feature's ``config.json`` setting is ``"EnableSearching ": false`` with options ``true`` and ``false``. |
+--------------------------------------------------------------------------------------------------------------+

> EnableAutocomplete

Author: ```suggestion
Enable Autocomplete
```

^^^^^^^^^^^^^^^^^^^^^^^
**True:** Autocomplete queries will use bleve search.

**False:** Autocomplete queries will not use bleve search. 
+-----------------------------------------------------------------------------------------------------------------+
| This feature's ``config.json`` setting is ``"EnableAutocomplete ": false`` with options ``true`` and ``false``. |
+-----------------------------------------------------------------------------------------------------------------+

> BulkIndexingTimeWindowSeconds

Author: ```suggestion
Bulk Indexing Time Window Seconds
```

| This feature's ``config.json`` setting is ``"Trace": ""`` with string input.                          |
+-------------------------------------------------------------------------------------------------------+

Bleve Settings (Experimental)
~~~~~~~~~~~~~~~

Index Dir
^^^^^^^^^^^^^^^^^^^^^^^
Directory path to use for storing bleve indexes.

+-----------------------------------------------------------------------------------------------------------+
| This feature's ``config.json`` setting is ``"IndexDir "`` with string input.                              |
+-----------------------------------------------------------------------------------------------------------+

Enable Indexing
^^^^^^^^^^^^^^^^^^^^^^^

> **True:** The indexing of new posts occurs automatically. Search queries will use database search until "Enable Bleve for search queries" is enabled.

Reviewer 2: nit: rewording to avoid referencing database as searches could be using Elasticsearch or other search engine in the future

```suggestion
**True:** The indexing of new posts occurs automatically. Search queries will not use bleve search until "Enable Bleve for search queries" is enabled.
```

**False:** The indexing of new posts does not occur automatically. 

+------------------------------------------------------------------------------------------------------------+
| This feature's ``config.json`` setting is ``"EnableIndexing": false`` with options ``true`` and ``false``. |
+------------------------------------------------------------------------------------------------------------+

Enable Searching
^^^^^^^^^^^^^^^^^^^^^^^
**True:** Search queries will use bleve search.

**False:** Search queries will not use bleve search.

+--------------------------------------------------------------------------------------------------------------+

> | This feature's ``config.json`` setting is ``"EnableSearching ": false`` with options ``true`` and ``false``. |

Reviewer 2: nit:
```suggestion
| This feature's ``config.json`` setting is ``"EnableSearching": false`` with options ``true`` and ``false``.  |
```

+--------------------------------------------------------------------------------------------------------------+

Enable Autocomplete
^^^^^^^^^^^^^^^^^^^^^^^
**True:** Autocomplete queries will use bleve search.

**False:** Autocomplete queries will not use bleve search. 

+-----------------------------------------------------------------------------------------------------------------+

> | This feature's ``config.json`` setting is ``"EnableAutocomplete ": false`` with options ``true`` and ``false``. |

Reviewer 2: nit:
```suggestion
| This feature's ``config.json`` setting is ``"EnableAutocomplete": false`` with options ``true`` and ``false``.  |
```

+-----------------------------------------------------------------------------------------------------------------+

Bulk Indexing Time Window Seconds
^^^^^^^^^^^^^^^^^^^^^^^
Determines the maximum time window for a batch of posts being indexed by the Bulk Indexer. This setting serves as a performance optimization for installs with over ~10 million posts in the database. Approximate this value based on the average number of seconds for 2,000 posts to be added to the database on a typical day in production. Setting this value too low will cause Bulk Indexing jobs to run slowly.

+-------------------------------------------------------------------------------------------------------------+

> | This feature's ``config.json`` setting is ``"BulkIndexingTimeWindowSeconds:3600  "`` with numerical input.  |

Reviewer 2: ```suggestion
| This feature's ``config.json`` setting is ``"BulkIndexingTimeWindowSeconds": 3600`` with numerical input.   |
```
